axon/src/lib.rs

```diff
@@ -212,13 +209,26 @@ pub fn rlp_opt_list<T: Encodable>(rlp: &mut RlpStream, opt: &Option<T>) {
 }
 
 pub fn convert_client_id_to_string(client_id: [u8; 32]) -> String {
-    let s = format!("{:x}", H256::from(client_id));
-    s
+    format!("{:x}", H256::from(client_id))
 }
 
-pub fn convert_string_to_client_id(s: &str) -> [u8; 32] {
-    let a = H256::from_str(s).unwrap().into();
-    a
+pub fn convert_string_to_client_id(s: &str) -> Result<[u8; 32], VerifyError> {
+    Ok(H256::from_str(s)
+        .map_err(|_| VerifyError::WrongClient)?
+        .into())
+}
+
+pub fn convert_connection_id_to_index(connection_id: &str) -> Result<usize, VerifyError> {
+    let index_str = connection_id
+        .split('-')
+        .last()
+        .ok_or(VerifyError::ConnectionsWrong)?;
+    let index = usize::from_str(index_str).map_err(|_| VerifyError::ConnectionsWrong)?;
+    Ok(index)
```

Comment:
This allows e.g. `invalid-8` or `connection-04`. Is this going to be a problem?

Reply:
yep, I put more detail on this